src/CSI.IBTA.UserService/Services/UsersService.cs

```diff
@@ -0,0 +1,301 @@
+            {
+                return new GenericResponse<UserDto>(true, new HttpError("Invalid User Role", HttpStatusCode.Unauthorized), null);
+            }
+            var employerUser = await _unitOfWork.EmployerUsers.Find(a => a.User.Id == user.Id);
+            int employerId = employerUser.Any() ? employerUser.First().Employer.Id : -1;
+            return new GenericResponse<UserDto>(false, null,
+                new UserDto(user.Id, user.Account.Username, user.Firstname, user.Lastname, user.Account.Id, employerId)
+                );
+        }
+
+        public async Task<GenericResponse<NewUserDto>> CreateUser(CreateUserDto createUserDto, HttpContext httpContext)
+        {
+            if (!IsNextSuperiorRole(httpContext.User, createUserDto.Role))
+            {
+                return new GenericResponse<NewUserDto>(true, new HttpError("Invalid User Role", HttpStatusCode.UnprocessableEntity), null);
+            }
+
+            var existingUser = await _unitOfWork.Users.Find(a => a.Account.Username == createUserDto.UserName);
+            
+            if (existingUser.Any())
+            {
+                return new GenericResponse<NewUserDto>(true, new HttpError("User already exists", HttpStatusCode.UnprocessableEntity), null);
+            }
+
+            var employer = await _unitOfWork.Employers.Find(a => a.Id == createUserDto.EmployerId);
+            if (!employer.Any())
+            {
+                return new GenericResponse<NewUserDto>(true, new HttpError("Employer not found", HttpStatusCode.NotFound), null);
+            }
+
+            Account newAccount = new Account()
+            {
+                Username = createUserDto.UserName,
+                Password = PasswordHasher.Hash(createUserDto.Password),
+                Role = createUserDto.Role
+            };
+
+            User newUser = new User()
+            {
+                Firstname = createUserDto.FirstName,
+                Lastname = createUserDto.LastName,
+                Account = newAccount,
+            };
+
+            Address newUserAddress = new Address()
+            {
+                State = createUserDto.AddressState,
+                Street = createUserDto.AddressStreet,
+                City = createUserDto.AddressCity,
+                Zip = createUserDto.AddressZip,
+                Account = newUser
+            };
+
+            Email newUserEmail = new Email()
+            {
+                EmailAddress = createUserDto.EmailAddress,
+                Account = newUser,
+            };
+
+            Phone newUserPhone = new Phone()
+            {
+                PhoneNumber = createUserDto.PhoneNumber,
+                Account = newUser,
+            };
+
+            EmployerUser newEmployerUser = new EmployerUser()
+            {
+                Employer = employer.First(),
+                User = newUser
+            };
+
+            await _unitOfWork.Accounts.Add(newAccount);
```

Comment:
User EF mappings, you should not manually solve relations. 
This note relates to the rest of the methods, I am not gonna repeat it, but all the methods of this class should be refactored. 
Ask colleagues to help and parallel the efforts if needed.